doc/manual/syntax.md

```diff
@@ -0,0 +1,446 @@
+```
+> let log = m#"
+if log:
+  print("log:", s)
+"#m in m#"
+def concat(str_list, log=false):
+  res = []
+  for s in str_list:
+    #{log}
+    res.append(s)
+  return res
+"#m
+"def concat(str_list, log=false):
+  res = []
+  for s in str_list:
+    if log:
+      print("log:", s)
+    res.append(s)
+  return res"
+```
+
+## Equality Operators
+
+We have seen that the operators `==` and `!=` works for comparing numbers, but they can actually compare any values together. To be equal, except for the obvious equality on values, elements must have the same type.
+
+Examples:
+```
+> 1 == 1
+true
+
+> 5 == 5.0
+true
+
+> "Hello" == "Hello"
+true
+
+> "Hello" != "World"
+true
+
+> 5 == "Hello"
+false
+
+> true == "true"
+false
+```
+
+
+## Composite values
+
+### List
+A list is a sequence of values. They are delimited by `[` and `]`, and elements are separated with `,`.
+
+Examples:
+```
+[1, 2, 3]
+["Hello", "World"]
+[1, true, "true"]
+[]
+```
+
+Lists can be concatenated with the operator `@`:
+```
+> [1] @ [2, 3]
+[ 1, 2, 3 ]
+```
+
+### Record
+Records are key-value storage, or in Nickel terms, field-value storage. They are delimited by `{` and `}`, and elements are separated by `,`.
+Field-value elements are noted as `field = value`.
+The fields are strings, but can be noted as identifiers if they respect identifiers syntax (internally, they will be stored as strings). Values can be of any type.
+Elements inside a record are unordered.
+Two records can be _merged_ together using the operator `&`. The reader can find more information about merging in the relevant documentation.
+
+Examples:
+```
+{}
+{a = 3}
+{my_id_n5 = "my id number 5", my_id_n4 = }
+{"5" = 5, six = 6}
+```
+
+Accessing a record field can be done using the `.` operator :
+```
+> { a = 1, b = 5 }.a
+1
+
+> { a = 1 }.b
+error: Missing field
+
+> { "1" = "one" }."1"
+"one"
+```
+
+It is possible to write records of records via the *piecewise syntax*, where we separate fields by dots:
+```
+> { a = { b = 1 } }
+{ a = { b = 1 } }
+ 
+> { a.b = 1 }
+{ a = { b = 1 } }
+
+> { a.b = 1, a.c = 2, b = 3}
+{ a = { b = 1, c = 2 }, b = 3 }
+```
+
+Since all fields are strings, we can use string interpolation to create keys or accessing:
+```
+> let k = "a" in { "#{k}" = 1 }
+{ a = 1 }
+
+> let k = "a" in { a = 1 }."#{k}"
+1
+```
+
+## Constructs
+
+### If-Then-Else
+This construct allows conditional branching in your code. You can use it as `if <bool expr> then <expr> else <expr>`.
+
+Examples:
+```
+> if true then "TRUE :)" else "false :("
+"TRUE :)"
+
+> if false then "Not this one" else "This one"
+"This one"
+
+> if "forty-two" == 42 then "equal?" else "unequal"
+"unequal"
+
+> ["1"] @ (if 42 == "42" then ["3"] else ["2"]) @ ["3"]
+["1", "2", "3"]
+```
+
+The `then` and `else` branches are only evaluated accordingly to the result of the expression inside the `if`.
```

Comment:
Sorry to insist, but I still don't understand the why of this paragraph. To me it is the very definition of an if-then-else, and as such it is obvious. Unless I'm misunderstanding something? What point do you want to make here, that would be true for Nickel but not for other languages?

Reply:
I don't see why it is obvious, especially for a configuration language, that someone might use without any prior knowledge of programming.

Reply:
meh, removed anyway